pkg/reconcilers/deploymentconfig.go

```diff
@@ -196,3 +196,16 @@ func DeploymentConfigPodTemplateLabelsMutator(desired, existing *appsv1.Deployme
 
 	return updated, nil
 }
+
+// DeploymentConfigEnvVarSyncMutator ensures pod env vars are reconciled
+func DeploymentConfigEnvVarSyncMutator(desired, existing *appsv1.DeploymentConfig) (bool, error) {
+	updated := false
+	for idx := range existing.Spec.Template.Spec.Containers {
+		if !reflect.DeepEqual(existing.Spec.Template.Spec.Containers[idx].Env, desired.Spec.Template.Spec.Containers[idx].Env) {
+			existing.Spec.Template.Spec.Containers[idx].Env = desired.Spec.Template.Spec.Containers[idx].Env
+			updated = true
+		}
+	}
+
+	return updated, nil
```

Comment:
Approach is probably too naive. Will prevent manually adding any new env vars to the deployment config 🤔